quarkus/src/main/java/com/redhat/openshift/knative/showcase/events/EventStore.java

```diff
@@ -0,0 +1,54 @@
+package com.redhat.openshift.knative.showcase.events;
+
+import io.cloudevents.CloudEvent;
+import io.smallrye.mutiny.Multi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+class EventStore {
+  private static final Logger LOGGER =
+    LoggerFactory.getLogger(EventStore.class);
+  private final List<CloudEvent> events = new ArrayList<>();
+  private final ExecutorService executorService =
+    Executors.newFixedThreadPool(12);
```

Comment:
I'm not 100% sure this implementation is valid.

Each connection should use the executor service to pull on the events. If none are present, we should wait a while, before checking once more. I feel this implementation, might block a thread per connection. Or not, as the `.runSubscriptionOn()` may interfere in this.

On the other hand, I don't know if we can avoid having a number of threads equal the maximum number of simultaneous connections...

More testing is needed...

Reply:
Definitely, this implementation means the number of threads equals the maximum number of simultaneous connections.

We can do better:

- Make an 'artificial' stream that emits a 'heartbeat' every 25ms and only then wakes up, looks at events and goes to sleep. That should handle any number of calls on a single thread
- Use a different implementation of events that is event-driven, that is, it tells you when something appears in it. Maybe using [`MultiCreate#emitter(java.util.function.Consumer)`](https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/MultiCreate.html#emitter(java.util.function.Consumer)) is a good idea. We can use a factory like this and just call emit on it as we get the next event. The stream emits nothing, and everything non-blocking sleeps.

Reply:
Yeah. Let's create an issue for this. AFAIK, this will not be used in production so this doesn't have to be perfect in the end.

Reply:
Tracked in https://github.com/openshift-knative/showcase/issues/16